internal/provider/validators.go

```diff
@@ -178,6 +178,40 @@ func validateConsulClusterSize(v interface{}, path cty.Path) diag.Diagnostics {
 	return diagnostics
 }
 
+func validateConsulClusterCIDR(v interface{}, path cty.Path) diag.Diagnostics {
```

Comment:
The Validations are pretty neat for `terraform plan`

<img width="678" alt="Screenshot 2023-02-10 at 1 08 36 AM" src="https://user-images.githubusercontent.com/23405655/218153497-e4bb542e-1a78-463d-995a-2e34a0667c7c.png">

<img width="1108" alt="Screenshot 2023-02-10 at 1 10 04 AM" src="https://user-images.githubusercontent.com/23405655/218153634-43f1144b-766a-45e8-8563-6f696bc88210.png">

Reply:
I think the more ideal user experience would be to check for duplicates at the plan stage as well. Unless that's something only the backend can validate?

Reply:
See this https://github.com/hashicorp/terraform-provider-hcp/pull/455/files/cf57462ccfb8949e2711f2df5f9b45bd66abfa38#r1102997970 comment. I can't add a validateDiagFunc for the schema type (schema.TypeList not supported)

Reply:
Let me know if you prefer another existing pattern to verify on plan, as I'm not yet very familiar with this repo!